### PR TITLE
[v2.0.3] Fixed Issue #3164 #3187 #3167

### PR DESCRIPTION
**Pull request by navya-BTC**
Fixed issue

- #3164 : added a success message
- #3167 : added delete icon
- #3187 : updated validation message

### CODE REVIEW

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp [hunk 1]
```diff
@@ -26,7 +31,7 @@
         </span>
         <span
             class="ml-xs sprites_v3 filled-tooltip" data-toggle="tooltip"
-            title="This must be a human-readable activity identifier and unique across all activities of the study.Note that this field cannot be edited once the study is Launched."></span>
+            title="A human-readable identifier that must be unique across all activities of the study. Allowed characters are lowercase alphabets (a-z), digits (0-9), _ (underscore) and -(minus)."></span>
```
Comment (moschetti) [inline study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp:34]: change to say `... are lowercase letters (a-z), ...` and add a space between the `-` & `(minus)` at the end.

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp [hunk 2]
```diff
@@ -27,7 +32,7 @@
       </span>
       <span
           class="ml-xs sprites_v3 filled-tooltip" data-toggle="tooltip"
-          title="This must be a human-readable activity identifier and unique across all activities of the study.Note that this field cannot be edited once the study is Launched."></span>
+          title="A human-readable identifier that must be unique across all activities of the study. Allowed characters are lowercase alphabets (a-z), digits (0-9), _ (underscore) and -(minus)."></span>
```
Comment (moschetti) [inline study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp:35]: same 2 changes as above

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp [hunk 3]
```diff
@@ -922,14 +927,14 @@
...
+            var regularExpression = /^[a-z0-9_-]*$/;
             if (!regularExpression.test($(this).val())) {
-              var newVal = $(this).val().replace(/[^A-Za-z0-9\*\(\)_+|:.\-]/g, '');
+              var newVal = $(this).val().replace(/[^a-z0-9_-]/g, '');
               e.preventDefault();
               $(this).val(newVal);
               $(this).parent().addClass("has-danger has-error");
               $(this).parent().find(".help-block").empty().append($("<ul><li> </li></ul>").attr("class","list-unstyled").text(
-                  "The characters like (< >) are not allowed."));
+                  "Please use allowed characters only: lowercase alphabets (a-z), digits (0-9), _ (underscore) and -(minus)."));
```
Comment (moschetti) [inline study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp:937]: same 2 changes as above

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp (hunk 1 above)
Comment (navya-BTC) [reply study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp]: Fixed review comment.

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp (hunk 2 above)
Comment (navya-BTC) [reply study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp]: Fixed review comment.

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp (hunk 3 above)
Comment (navya-BTC) [reply study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewTowerStudyActiveTask.jsp]: Fixed review comment.

File: study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp [hunk 4]
```diff
@@ -2324,14 +2332,14 @@
...
+        var regularExpression = /^[a-z0-9_-]*$/;
         if (!regularExpression.test($(this).val())) {
-          var newVal = $(this).val().replace(/[^A-Za-z0-9\*\(\)_+|:.\-]/g, '');
+          var newVal = $(this).val().replace(/[^a-z0-9_-]/g, '');
           e.preventDefault();
           $(this).val(newVal);
           $(this).parent().addClass("has-danger has-error");
           $(this).parent().find(".help-block").empty().append($("<ul><li> </li></ul>").attr("class","list-unstyled").text(
-              "The characters like (< >) are not allowed."));
+        		  "Please use allowed characters only: lowercase alphabets (a-z), digits (0-9), _ (underscore) and -(minus)."));
```
Comment (moschetti) [inline study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp:2342]: sorry, same change here: `lowercase letters` and space next to the `-`

Comment (navya-BTC) [reply study-builder/fdahpStudyDesigner/src/main/webapp/WEB-INF/view/study/viewSpatialStudyActiveTask.jsp]: Fixed review comment.

**Review by moschetti — APPROVED** [review]

**Pull request merged**